kustomize/commands/edit/remove/removebuildmetadata.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package remove
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type removeBuildMetadataOptions struct {
+	buildMetadataOptions []string
+}
+
+// newCmdRemoveBuildMetadata removes options to the kustomization's buildMetada field.
+func newCmdRemoveBuildMetadata(fSys filesys.FileSystem) *cobra.Command {
+	var o removeBuildMetadataOptions
+
+	cmd := &cobra.Command{
+		Use:   "buildmetadata",
+		Short: "Removes one or more buildMetadata options to the kustomization.yaml in the current directory",
+		Long: `Removes one or more buildMetadata options to the kustomization.yaml in the current directory.
+The following options are valid:
+  - originAnnotations
+  - transformerAnnotations
+  - managedByLabel
+originAnnotations will remove the annotation config.kubernetes.io/origin to each resource, describing where 
+each resource originated from.
+transformerAnnotations will remove the annotation alpha.config.kubernetes.io/transformations to each resource,
+describing the transformers that have acted upon the resource.
+managedByLabel will remove the label app.kubernetes.io/managed-by to each resource, describing which version
+of kustomize managed the resource.`,
+		Example: `
+		remove buildmetadata {option1},{option2}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+			return o.RunRemoveBuildMetadata(fSys)
+		},
+	}
+	return cmd
+}
+
+// Validate validates removeBuildMetadata command.
+func (o *removeBuildMetadataOptions) Validate(args []string) error {
+	if len(args) == 0 {
+		return errors.New("must specify a buildMetadata option")
+	}
+	if len(args) > 1 {
+		return fmt.Errorf("too many arguments: %s; to provide multiple buildMetadata options, please separate options by comma", args)
+	}
+	o.buildMetadataOptions = strings.Split(args[0], ",")
+	for _, opt := range o.buildMetadataOptions {
+		if !kustfile.StringInSlice(opt, types.BuildMetadataOptions) {
+			return fmt.Errorf("invalid buildMetadata option: %s", opt)
+		}
+	}
+	return nil
+}
+
+// RunRemoveBuildMetadata runs removeBuildMetadata command (do real work).
+func (o *removeBuildMetadataOptions) RunRemoveBuildMetadata(fSys filesys.FileSystem) error {
+	mf, err := kustfile.NewKustomizationFile(fSys)
+	if err != nil {
+		return err
+	}
+	m, err := mf.Read()
+	if err != nil {
+		return err
+	}
+	var newOptions []string
+	for _, opt := range m.BuildMetadata {
+		if !kustfile.StringInSlice(opt, o.buildMetadataOptions) {
+			newOptions = append(newOptions, opt)
+		}
+	}
+	m.BuildMetadata = newOptions
```

Comment:
It seems that the code doesn't provide a way to remove the `buildmetadata` field once all options are removed. Right? For example,

Before

```yaml
# kustomzation.yaml
buildMetadata: ["originAnnotations"]
namespace: test
```
```bash
kustomize edit remove buildmetadata originAnnotations
``` 
After (Actual)
```yaml
# kustomzation.yaml
buildMetadata: []
namespace: test
```
After (Expected)
```yaml
# kustomzation.yaml
namespace: test
```

Reply:
Fixed this! If buildMetadata is empty it is omitted from the kustomization